src/jit/lsraxarch.cpp

```diff
@@ -754,7 +754,9 @@ void LinearScan::BuildCheckByteable(GenTree* tree)
         if (tree->OperIsSimple())
         {
             GenTree* op = tree->gtOp.gtOp1;
-            if (op != nullptr)
+            // We need byte registers on the operands of most simple operators that produce a byte result.
+            // However, indirections are simple operators but do not require their address in a byte register.
```

Comment:
In which cases does indirection produce byte results?

Reply:
If the indirection is loading a byte from memory (e.g. a field of a struct), it will be of type byte. The register it loads into must be a byte register, but the address need not be.

Reply:
Hmm, usually an indirection of type byte/ubyte will generate a movsx/movzx instruction. In that case you don't need a byte register, the destination register is really a 32 bit register.